src/updates/fasta.rs

```diff
@@ -69,73 +69,113 @@ pub fn update_with_fasta(
+
+            BlockGroup::insert_change(conn, &path_change, &interval_tree).unwrap();
+            if index == 0 {
+                first_node = Some(node_id);
+            } else if first_node.is_some() {
+                first_node = None;
+            }
+        } else {
+            let seq = Sequence::new()
+                .sequence_type("DNA")
+                .sequence(&sequence)
+                .save(conn);
+            let node_id = Node::create(
+                conn,
+                &seq.hash,
+                &HashId::convert_str(&format!(
+                    "{path_id}:{ref_start}-{ref_end}->{sequence_hash}",
+                    path_id = path.id,
+                    ref_start = 0,
+                    ref_end = seq.length,
+                    sequence_hash = seq.hash
+                )),
+            );
+
+            let path_block = PathBlock {
+                id: -1,
+                node_id,
+                block_sequence: sequence,
+                sequence_start: 0,
+                sequence_end: seq.length,
+                path_start: start_coordinate,
+                path_end: end_coordinate,
+                strand: Strand::Forward,
+            };
+
+            let path_change = PathChange {
+                block_group_id: new_block_group_id,
+                path: path.clone(),
+                path_accession: None,
+                start: start_coordinate,
+                end: end_coordinate,
+                block: path_block,
+                chromosome_index: NO_CHROMOSOME_INDEX,
+                phased: 0,
+                preserve_edge: true,
+            };
+
+            BlockGroup::insert_change(conn, &path_change, &interval_tree).unwrap();
+            if index == 0 {
+                first_node = Some(node_id);
+            } else if first_node.is_some() {
+                first_node = None;
+            }
         }
+
         change_count += 1;
     }
 
     if !disable_reference_path_update && let Some(node_id) = first_node {
-        let edge_to_new_node = Edge::query(
-            conn,
-            "select * from edges where target_node_id = ?1",
-            rusqlite::params!(SQLValue::from(node_id)),
-        )[0]
-        .clone();
-        let edge_from_new_node = Edge::query(
-            conn,
-            "select * from edges where source_node_id = ?1",
-            rusqlite::params!(SQLValue::from(node_id)),
-        )[0]
-        .clone();
-        path.new_path_with(
-            conn,
-            start_coordinate,
-            end_coordinate,
-            &edge_to_new_node,
-            &edge_from_new_node,
-        );
+        if node_id == HashId::convert_str("") {
+            let _ = path.new_path_with_deletion(conn, start_coordinate, end_coordinate);
+        } else {
+            let edge_to_new_node = Edge::query(
+                conn,
+                "select * from edges where target_node_id = ?1",
+                rusqlite::params!(SQLValue::from(node_id)),
```

Comment:
you can actually rewrite this just as `params![node_id]`, no need to import SQLValue.